File: _wpf/wpf-business-programmer/3-wpf-business-programmer-m3-exercise-files/Samples-2012/VB/WPFBusinessSamples/WPFComponents/Address.vb
Imports System.Text

Public Class Address
  Public Property Street1 As String
  Public Property Street2 As String
  Public Property City As String
  Public Property State As String
  Public Property PostalCode As String

  Public Function GetAddress() As String
    Dim sb As New StringBuilder()

    sb.Append(Street1 & Environment.NewLine)
    sb.Append(Street2 & Environment.NewLine)
    sb.Append(City & ", " & State & "  " & PostalCode)

    Return sb.ToString()
  End Function

  Public Function CreateSample() As Address
    Dim addr As New Address()

    addr.Street1 = "17852 E. 17th Street"
    addr.Street2 = "Suite 205"
    addr.City = "Tustin"
    addr.State = "CA"
    addr.PostalCode = "92780"

    Return addr
  End Function
End Class

File: _wpf/wpf-business-programmer/3-wpf-business-programmer-m3-exercise-files/Samples/CSharp/WPFBusinessSamples/WPFComponents/Address.cs
using System;
using System.Text;

namespace WPFComponents
{
  public class Address
  {
    public string Street1 { get; set; }
    public string Street2 { get; set; }
    public string City { get; set; }
    public string State { get; set; }
    public string PostalCode { get; set; }

    public string GetAddress()
    {
      StringBuilder sb = new StringBuilder();

      sb.Append(Street1 + Environment.NewLine);
      sb.Append(Street2 + Environment.NewLine);
      sb.Append(City + ", " + State + "  " + PostalCode);

      return sb.ToString();
    }

    public Address CreateSample()
    {
      Address addr = new Address();

      addr.Street1 = "17852 E. 17th Street";
      addr.Street2 = "Suite 205";
      addr.City = "Tustin";
      addr.State = "CA";
      addr.PostalCode = "92780";

      return addr;
    }
  }
}
